contracts/erc20-gateway/ERC20Cogateway.sol

```diff
@@ -203,4 +218,103 @@ contract ERC20Cogateway is
             messageHash_
         );
     }
+
+    /**
+     * @notice Confirm deposit in order to mint tokens.
+     *
+     * @param _amount ERC20 token deposit amount in atto.
+     * @param _beneficiary Address of beneficiary on target metachain.
+     * @param _feeGasPrice Gas price at which fee will be calculated.
+     * @param _feeGasLimit Gas limit at which fee will be capped.
+     * @param _valueToken ERC20 token address.
+     * @param _depositor Address of depositor on the origin chain.
+     * @param _blockNumber Block number of origin chain against which storage
+                           proof is generated.
+     * @param _rlpParentNodes Storage merkle proof to verify message declaration
+     *                        on the origin chain.
+     *
+     * @return messageHash_ Message hash.
+     */
+    function confirmDeposit(
+        uint256 _amount,
+        address payable _beneficiary,
+        uint256 _feeGasPrice,
+        uint256 _feeGasLimit,
+        address _valueToken,
+        address _depositor,
+        uint256 _blockNumber,
+        bytes calldata _rlpParentNodes
+     )
+        external
+        returns (bytes32 messageHash_)
+    {
+        uint256 initialGas = gasleft();
+
+        require(
+            _amount != 0,
+            "Deposit amount should be greater than 0."
+        );
+        require(
+            _beneficiary != address(0),
+            "Beneficiary address must not be 0."
+        );
+        require(
+            _depositor != address(0),
+            "Depositor address must not be 0."
+        );
+
+        deployUtilityToken(_valueToken);
```

Comment:
it will not always deploy utility token. Either rename to `tryDeployUtilityToken` or pullout if condition and call `deployUtilityToken`.

Reply:
I renamed it to `getUtilityToken`